### DISCUSSION TITLE
How to handle navigationDelegate of webView object.

### DISCUSSION THREAD

**Discussion by kythanh** [kylehickinson/SwiftUI-WebView#8] [Q&A]
Hi @kylehickinson 
Do you have any idea how to handle navigationDelegate of webView object. Below are one of delegate I would like to handle with the webView object. Thank you.

```
public func webView(_ webView: WKWebView, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
            if challenge.protectionSpace.host == "correcturl.com" {
                let user = "user"
                let password = "password"
                let credential = URLCredential(user: user, password: password, persistence: URLCredential.Persistence.forSession)
                challenge.sender?.use(credential, for: challenge)
                completionHandler(.useCredential, credential)
            }
    }
```

**Comment by kylehickinson** [comment]
@kythanh There's a few different way's you can do this. `WebViewStore` accepts handling a specific `WKWebView` that you pass through its initializer, so you could just create a `WKWebView` elsewhere, set up its delegates and hand it off to the `WebViewStore`. Otherwise you can always manually set the delegate's on the managed web view created by `WebViewStore` as well.

**Reply by kythanh** [reply]
Hi @kylehickinson 
Thank for you tip. I tried this one https://github.com/kylehickinson/SwiftUI-WebView/commit/3a2f0603bed9bef2e647a116dc1c37550178dd60, it's working for me but not sure if that is best practice as I am new to SwiftUI. Do you have any advise?

**Reply by kylehickinson** [reply]
That's certainly a way to do it if you plan on forking and editing the library itself, it just means if you need any customization you'll be passing in those details to the `WebView` instead of the `WebViewStore` or directly on whatever store is holding a reference to the `WKWebView` :)
